mkt/api/paginator.py

```diff
@@ -60,61 +60,55 @@ def page(self, number):
-        request_data.update(params)
-        query = urlencode(request_data)
-        return urlparse.urlunsplit((scheme, netloc, path, query, fragment))
-
-    def get_offset_link_for_page(self, page, number):
-        request = self.context.get('request')
-        url = request and request.get_full_path() or ''
-        number = number - 1  # Pages are 1-based, but offsets are 0-based.
-        per_page = page.paginator.per_page
-        params = {'offset': number * per_page, 'limit': per_page}
-        request_data = request and request.GET.dict() or {}
-        return self.replace_query_params(url, request_data, params)
-
-    def get_next(self, page):
-        if not page.has_next():
-            return None
-        return self.get_offset_link_for_page(page, page.next_page_number())
-
-    def get_previous(self, page):
-        if not page.has_previous():
+    count_override = 0
+
+    def __init__(self, default_limit=None):
+        pagination.LimitOffsetPagination.__init__(self)
+        if default_limit is not None:
+            self.default_limit = default_limit
+
+    def get_paginated_response(self, data):
+        return response.Response(OrderedDict([
+            ('meta', OrderedDict([
+                ('next', self.get_next_link()),
+                ('previous', self.get_previous_link()),
+                ('total_count', self.count),
+                ('offset', self.get_offset(self.request)),
+                ('limit', self.get_limit(self.request))])),
+            ('objects', data)]))
+
+    def paginate_queryset(self, queryset, request, view=None):
+        self.limit = self.get_limit(request)
+        if self.limit is None:
             return None
-        return self.get_offset_link_for_page(page, page.previous_page_number())
-
-    def get_total_count(self, page):
-        return page.paginator.count
-
-    def get_offset(self, page):
-        index = page.start_index()
-        if index > 0:
-            # start_index() is 1-based, and we want a 0-based offset, so we
-            # need to remove 1, unless it's already 0.
-            return index - 1
-        return index
-
-    def get_limit(self, page):
-        return page.paginator.per_page
-
 
-class CustomPaginationSerializer(pagination.BasePaginationSerializer):
-    meta = MetaSerializer(source='*')  # Takes the page object as the source
-    results_field = 'objects'
+        self.offset = self.get_offset(request)
+        self.count = self.count_override or pagination._get_count(queryset)
+        self.request = request
+        if self.count > self.limit and self.template is not None:
+            self.display_page_controls = True
+        page = queryset[self.offset:self.offset + self.limit]
+        if isinstance(page, Search):
+            page = page.execute()
+        return list(page)
+
+
+class PageNumberPagination(pagination.PageNumberPagination):
+
+    def django_paginator_class(self, queryset, page_size):
```

Comment:
You removed `ESPaginator` from some views that do ES queries, and `ESPaginator` was useful for those because it prevented the extra count query in ES. So if `LimitOffsetPagination` does not use `ESPaginator`, then the extra count query is going to take place, which is bad. We need a way to fix that.